rgw/v2/tests/s3cmd/test_large_object_gc.py

```diff
@@ -0,0 +1,181 @@
+"""
+test_s3cmd - Test large object download with gc using s3cmd
+
+Usage: test_large_object_gc.py -c <input_yaml>
+
+<input_yaml>
+    Note: Following yaml can be used
+    test_large_object_gc.yaml
+
+Operation:
+    Create an user
+    Create a bucket with user credentials
+    Upload large file to bucket
+    Set rgw_gc_obj_min_wait as 5
+    Download uploaded object
+    Verify download is succeeded
+    Delete uploaded object
+    Delete bucket
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import socket
+import sys
+import time
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib import resource_op
+from v2.lib.admin import UserMgmt
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.lib.s3cmd import auth as s3_auth
+from v2.tests.s3_swift import reusable
+from v2.tests.s3cmd import reusable as s3cmd_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+
+
+def test_exec(config):
+    """
```

Comment:
Would it be possible to break this method into smaller pieces? I believe it would help in re-usability and maintenance

Reply:
@psathyan , I have done in the same way of existing tests in repo to make it similar to other tests. Can we plan this as separate PR where we can reformat and reuse for all the applicable tests where the initial steps for almost same?

Reply:
most of the reusable functions are placed in `reusables.py`. each test case will have its own unique tests steps. @vivekanandan-k-rh can you check if there is a possibility of adding more functions from here to reusable

Reply:
IMO, we could break into smaller methods instead of having a single method. Doing the work now would avoid rework as it is a new module that is being added.

Reply:
@psathyan Sure. I'll break it to maximum possible methods and update the PR.

Reply:
@psathyan , I have created new method get_rgw_ip_and_port by moving the codes which which will be used in most of the further cases to be added. I believe other codes are calling already existing methods in reusable.py and other files. Please review the same.